### PR TITLE
Lists component improvements

### PR DESCRIPTION
**Pull request by mbmjertan**
This PR:
* implements a way to change the value of the `listsOrdered` attribute from the editor
* implements additional checks that the `listsOrdered` attribute can be only `ul` or `ol`
  * alternatively, we could make the `listsOrdered` attribute a boolean, but I decided for this approach to not break backwards compatibility - let me know if you think we should have a different approach

### CODE REVIEW

File: blocks/init/src/Blocks/components/lists/components/lists-options.js [hunk 1]
```diff
@@ -62,6 +64,18 @@ export const ListsOptions = (attributes) => {
 						/>
 					}
 
+					{showListsOrdered &&
+						<CustomSelect
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/components/lists-options.js:68]: As this is less than 5 options, a `SimpleVerticalSingleSelect` would be a better option here.

There are icons for (un)ordered lists you can use 😉 
image

File: blocks/init/src/Blocks/components/lists/manifest.json [hunk 2]
```diff
@@ -38,6 +38,16 @@
 		}
 	},
 	"options": {
+		"listsOrdered": [
+			{
+				"label": "Ordered list",
+				"value": "ol"
+			},
+			{
+				"label": "Bullet list",
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/manifest.json:47]:
```suggestion
				"label": "Bulleted list",
```

File: blocks/init/src/Blocks/components/lists/components/lists-editor.js [hunk 3]
```diff
@@ -25,6 +25,12 @@ export const ListsEditor = (attributes) => {
 	const listsContent = checkAttr('listsContent', attributes, manifest);
 	const listsOrdered = checkAttr('listsOrdered', attributes, manifest);
 
+	const listsOrderedOptions = getOption('listsOrdered', attributes, manifest).map(option => option.value);
+	const listsDefault = manifest.attributes.listsOrdered.default;
+	if (!listsOrderedOptions.includes(listsOrdered)) {
+		setAttributes({ [getAttrKey('listsOrdered', attributes, manifest)]: listsDefault }); // Resets the attribute to the default value if invalid value set.
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/components/lists-editor.js:31]: Is there a way to force an invalid value besides manually editing the code view?

If not, maybe this check should be only on PHP side

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Left a couple of Qs and suggestions, nice job!

File: blocks/init/src/Blocks/components/lists/components/lists-editor.js (hunk 3 above)
Comment (mbmjertan) [reply blocks/init/src/Blocks/components/lists/components/lists-editor.js:31]: There isn't, but `RichText` is rendered with that HTML tag, so I'm not too sure we should render an attribute we're not sure is safe.

File: blocks/init/src/Blocks/components/lists/components/lists-options.js [hunk 4]
```diff
@@ -62,6 +64,18 @@ export const ListsOptions = (attributes) => {
 						/>
 					}
 
+					{showListsOrdered &&
+						<SimpleVerticalSingleSelect
+							label={__('List type', 'edp')}
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/components/lists-options.js:69]:
```suggestion
							label={__('List type', 'eightshift-frontend-libs')}
```

File: blocks/init/src/Blocks/components/lists/manifest.json [hunk 5]
```diff
@@ -38,6 +38,18 @@
 		}
 	},
 	"options": {
+		"listsOrdered": [
+			{
+				"label": "Ordered list",
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/manifest.json:43]:
```suggestion
				"label": "Ordered",
```

File: blocks/init/src/Blocks/components/lists/manifest.json [hunk 6]
```diff
@@ -38,6 +38,18 @@
...
+				"value": "ol",
+				"icon": "listOrdered"
+			},
+			{
+				"label": "Bulleted list",
```
Comment (goranalkovic-infinum) [inline blocks/init/src/Blocks/components/lists/manifest.json:48]:
```suggestion
				"label": "Bulleted",
```

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Couple more for cohesion 😄

**Review by dingo-d — DISMISSED** [review]
Just fix what Goc suggested and it's ok from my end 👍🏼

**Review by goranalkovic-infinum — APPROVED** [review]

**Review by simun-ivanac — APPROVED** [review]

**Pull request merged**